### PR TITLE
Alerting: Receivers integrations error feedback: WIP - Add notifications error at the top right on contact points view

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What this PR does / why we need it**: 
This PR is a WIP, and prepares the FE to show the number of channel contact points that are failing in the contact points view. In this PR we are using a 'fake' response as we still don't have the new endpoint to get these errors.

Screenshot 2022-07-18 at 11 57 03

### CODE REVIEW

File: public/app/types/alerting.ts [hunk 1]
```diff
@@ -142,6 +142,36 @@ export interface NotificationChannelState {
   notificationChannel: any;
 }
 
+export interface IntegrationError {
+  lastError: string;
+  lastNotify: DateTime;
```
Comment (konrad147) [inline public/app/types/alerting.ts:147]: Does it come from the `API`? If yes, then this should probably be of type `string`

File: public/app/features/alerting/unified/Receivers.tsx [hunk 2]
```diff
@@ -68,12 +106,33 @@ const Receivers: FC = () => {
...
+        <AlertManagerPicker
+          current={alertManagerSourceName}
+          disabled={disableAmSelect}
+          onChange={setAlertManagerSourceName}
+          dataSources={alertManagers}
+        />
+        {integrationsErrorCount > 0 && (
+          <NotificationError>
+            <div
+              className={css`
```
Comment (konrad147) [inline public/app/features/alerting/unified/Receivers.tsx:124]: Please, create a `getStyles` function for the styles in this component. The same as for the `NotificationError` component

File: public/app/features/alerting/unified/Receivers.tsx [hunk 3]
```diff
@@ -29,12 +49,17 @@ const Receivers: FC = () => {
...
   const configRequests = useUnifiedAlertingSelector((state) => state.amConfigs);
+  const contactPointsStateRequest = useUnifiedAlertingSelector((state) => state.contactPointsState);
 
   const {
     result: config,
     loading,
     error,
   } = (alertManagerSourceName && configRequests[alertManagerSourceName]) || initialAsyncRequestState;
+
+  const { result: contactPointsState } =
```
Comment (konrad147) [inline public/app/features/alerting/unified/Receivers.tsx:60]: This is a bit unclear. Do we expect to get the same response from two different requests?

**Review by konrad147 — COMMENTED** [review]
Nice the first PR! 🙌  Just a few nits

File: public/app/types/alerting.ts (hunk 1 above)
Comment (soniaAguilarPeiron) [reply public/app/types/alerting.ts:147]: do you mean lastNotify?, yes

Comment (konrad147) [reply public/app/types/alerting.ts]: Yes, I mean `lastNotify`. It is a JSON object deserialized from the server response, so the type will be rather `string` than `DateTime`

File: public/app/features/alerting/unified/Receivers.tsx (hunk 3 above)
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/Receivers.tsx:71]: initialAsyncRequestState is the default value

File: public/app/features/alerting/unified/Receivers.tsx [hunk 4]
```diff
@@ -16,25 +19,58 @@ import { ReceiversAndTemplatesView } from './components/receivers/ReceiversAndTe
...
 
+export interface NotificationErrorProps {
+  errorCount: number;
+}
+
+const NotificationError: FC<NotificationErrorProps> = ({ errorCount }: NotificationErrorProps) => {
+  const styles = useStyles2(getStyles);
+
+  return (
+    <div className={styles.warning}>
```
Comment (konrad147) [inline public/app/features/alerting/unified/Receivers.tsx:39]: Nice improvement! I like having everything error-related in a single component. 
You may also try the [Stack](https://github.com/grafana/grafana-experimental/blob/main/src/query-builder/Stack.tsx) component which helps to work with `flex` layouts.

Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/Receivers.tsx:39]: ohh!!! thank you @konrad147 !!! I will take a look! 👀

Comment (konrad147) [reply public/app/features/alerting/unified/Receivers.tsx:40]: Nice!

**Review by konrad147 — APPROVED** [review]
Looks nice! 👏

**Pull request merged**